app/db/repository/authenticator_repository.rb

```diff
@@ -5,34 +5,74 @@ def initialize(resource_repository: ::Resource)
         @resource_repository = resource_repository
       end
 
-      def find(type:, account:, service_id:)
+      def find_all(type:, account:, role: nil)
+        args_list = []
+        variables = fetch_authenticators(account: account, type: type, service_id: nil, role: role)
+        variables.each do |variable|
+          next unless variable.secret
+
+          args = {}
+          args[:service_id] = variable.owner_id.split('/')[-1].underscore.to_sym
+          args[:account] = account
+          args[variable.resource_id.split('/')[-1].underscore.to_sym] =
+            variable.secret.value
+          args_list.push(args)
+        end
+        args_list.group_by{|authn| authn[:service_id]}.map do |_, authn|
+          "Authenticator::#{type.camelize}Authenticator".constantize.new(**authn.reduce({}, :merge))
+        end
+      end
+
+      def find(type:, account:, service_id:, role: nil)
         return nil unless exists?(
           type: type,
           account: account,
           service_id: service_id
         )
 
-        variables = @resource_repository.where(
-          Sequel.like(
-            :resource_id,
-            "#{account}:variable:conjur/authn-#{type}/#{service_id}/%"
-          )
-        ).eager(:secrets).all
+        variables = fetch_authenticators(account: account, type: type, service_id: service_id, role: role)
 
         args_list = {}.tap do |args|
           args[:account] = account
           args[:service_id] = service_id
           variables.each do |variable|
             next unless variable.secret
+
             args[variable.resource_id.split('/')[-1].underscore.to_sym] = variable.secret.value
           end
         end
 
         "Authenticator::#{type.camelize}Authenticator".constantize.new(**args_list)
       end
 
+      def resources(role:)
+        unless role
+          return @resource_repository
```

Comment:
Won't this bypass our security?  If the role isn't present, it'll return ALL resources instead of those which are visible to the requester.